lightning/src/chain/channelmonitor.rs

```diff
@@ -782,7 +626,7 @@ pub struct ChannelMonitor<ChanSigner: ChannelKeys> {
 	/// spending. Thus, in order to claim them via revocation key, we track all the counterparty
 	/// commitment transactions which we find on-chain, mapping them to the commitment number which
 	/// can be used to derive the revocation key and claim the transactions.
-	counterparty_commitment_txn_on_chain: HashMap<Txid, (u64, Vec<Script>)>,
+	counterparty_commitment_txn_on_chain: HashMap<Txid, u64>,
```

Comment:
Can this map be dropped now?  The commitment number can be decoded from the commitment TX via:

```rust
        let commitment_number = (((commitment_tx.input[0].sequence as u64 & 0xffffff) << 3 * 8)
            | (commitment_tx.lock_time as u64 & 0xffffff))
            ^ obscure_factor;
```

Reply:
oh, I guess this is for sending watchtowers?

Reply:
Its used to figure out the commitment number given an HTLC tx (ie by looking at the previous output spent), which I don't believe is otherwise possible. Any suggestions for updating the comment to make that more clear?

Reply:
I must be missing something.  We cannot see HTLC transactions on-chain before we see the commitment tx on-chain and there can only be one of those.  Once we see the commitment tx, we can decode and stash the commitment number in variable in this struct, so we have it for the HTLC tx.  So I still don't see the need for a map.

Reply:
Right, there was some Slack discussion on this, but, yes, ultimately we could track only the latest remote commitment tx we saw on-chain (because to invalidate that we'd need a reorg, invalidating child HTLC txn too). The code to do that would likely be somewhat simpler, but I'd really like more testing in ChannelMonitor, so am ~0/-0 on such a change, at least not gonna jump to implement it myself.

Reply:
Thank you for summarizing the discussion, I might give it a try at some point.

Reply:
fyi @devrandom this map was introduced before we had any serious reorg-handling in ChannelMonitor and thus it was simpler at that time to just have a dumb map instead of stashing reorg'ed commitment.

A real improvement would be instead to store a tuple of (commitment_txid, per_commitment_point, per_commitment_key) as it would avoid to fetch the secret holder and re-derive a per-commitment each time we see a revoked child HTLC. It could be done once for all when we see the commitment. I think.

Reply:
Also I think you can remove the first line of comment "We cannot identify HTLC-Success or HTLC-Timeout transactions by themselves on the chain." In fact we should be able to do so if we watch well commitment outputs. There is a line of further parsing in `check_spend_counterparty_htlc` to qualify them well as HTLCs.

Reply:
I'm a little confused by the second comment there? Its true we can't identify them "by themselves", that's why we have the map, no?